tests/Tests/Unit/ApiExceptionTest.php

```diff
@@ -203,10 +204,105 @@ public function testCreateFromApiResponse($metadata, $metadataArray) {
         $this->assertSame($metadata, $apiException->getMetadata());
     }
 
+    public function getRestMetadata()
+    {
+        $unknownBinData = [
+            [
+                '@type' => 'unknown-bin',
+                'data' => '<Unknown Binary Data>'
+            ]
+        ];
+        $asciiData = [
+            [
+                '@type' => 'ascii',
+                'data' => 'ascii-data'
+            ]
+        ];
+        $retryInfoData = [
+            [
+                '@type' => 'google.rpc.retryinfo-bin',
+                'retryDelay' => [
+                    'seconds' => 1,
+                    'nanos' => 2,
+                ],
+            ]
+        ];
+        $allKnownTypesData = [
+            [
+                '@type' => 'google.rpc.retryinfo-bin',
+            ],
+            [
+                '@type' => 'google.rpc.debuginfo-bin',
+                "stackEntries" => [],
+                "detail" => ""
+            ],
+            [
+                '@type' => 'google.rpc.quotafailure-bin',
+                'violations' => [],
+            ],
+            [
+                '@type' => 'google.rpc.badrequest-bin',
+                'fieldViolations' => []
+            ],
+            [
+                '@type' => 'google.rpc.requestinfo-bin',
+                'requestId' => '',
+                'servingData' => '',
+            ],
+            [
+                '@type' => 'google.rpc.resourceinfo-bin',
+                'resourceType' => '',
+                'resourceName' => '',
+                'owner' => '',
+                'description' => '',
+            ],
+            [
+                '@type' => 'google.rpc.help-bin',
+                'links' => [],
+            ],
+            [
+                '@type' => 'google.rpc.localizedmessage-bin',
+                'locale' => '',
+                'message' => '',
+            ],
+        ];
+
+        return [
+            [
```

Comment:
Add a null case for metadata, since the function allows for it?
```suggestion
            [
                [null],
```

Reply:
Ah sorry I thought that an empty array (the first test) was testing for null, but I see now that `[]` is different from `[null]`. Done.